### ISSUE TITLE
Unable to highlight users in bold text and italics

### ISSUE THREAD

**Issue by paulchen** [RocketChat/Rocket.Chat#28843]
### Description:

When formatting text in bold face or italics, users will not be highlighted.

### Steps to reproduce:

1. Compose a message highlighting some user.
2. Before submitting the message, format the message in bold face or italics.
3. Submit the message.

### Expected behavior:

The user will be highlighted.

### Actual behavior:

The user is not highlighted.

![image](https://user-images.githubusercontent.com/404840/230746563-d9a0c35f-be99-4fa9-b7d0-7ddbf327405e.png)

### Server Setup Information:

- Version of Rocket.Chat Server: 6.1.1
- Deployment Method: Docker
- Number of Running Instances: 1
- MongoDB Version: 6.0.5

### Client Setup Information

Issue is client-agnostic.

### Additional context

This also affects messages composed using `/me` as these messages are automatically formatted in italics.

**Comment by saurabhmj11** [comment]
The issue you are experiencing with formatting text in bold or italics and not being able to highlight users could be related to the client or server setup. Here are some steps you can take to troubleshoot this issue:

Check if the user highlighting feature is enabled in the server settings. To do this, go to the "Administration" panel and look for the "Message Highlighting" option. Make sure it is enabled.

Check if the issue occurs in different clients. Try using a different client, such as the desktop or mobile app, and see if the same issue occurs.

Check if the issue occurs in different browsers. Try using a different browser and see if the same issue occurs.

Check if there are any conflicting custom CSS rules that may be interfering with the user highlighting feature. Try disabling any custom CSS rules and see if the issue is resolved.

Check if the issue is related to the formatting syntax used. Try using a different formatting syntax, such as Markdown, and see if the same issue occurs.

Check if the issue is related to the version of Rocket.Chat Server. Try upgrading to the latest version of Rocket.Chat Server and see if the issue is resolved.

If none of these steps resolve the issue, you may want to reach out to the Rocket.Chat community for additional support or file a bug report if it is a software issue.

**Comment by paulchen** [comment]
> Check if the user highlighting feature is enabled in the server settings. To do this, go to the "Administration" panel and look for the "Message Highlighting" option. Make sure it is enabled.

No, I won't do that as this cannot be the reason for the issue. Highlighting works _in general_, only _under certain circumstances_ it does not.

> Check if the issue occurs in different clients. Try using a different client, such as the desktop or mobile app, and see if the same issue occurs.

I did that. It occurs everywhere. That's why I wrote that the issue is "client-agnostic".

> Check if the issue occurs in different browsers. Try using a different browser and see if the same issue occurs.

I did that. It occurs everywhere. That's why I wrote that the issue is "client-agnostic".

> Check if there are any conflicting custom CSS rules that may be interfering with the user highlighting feature. Try disabling any custom CSS rules and see if the issue is resolved.

No custom CSS rules are enabled.

> Check if the issue is related to the formatting syntax used. Try using a different formatting syntax, such as Markdown, and see if the same issue occurs.

I don't know of another formatting syntax for bold face or italics besides Markdown.

> Check if the issue is related to the version of Rocket.Chat Server. Try upgrading to the latest version of Rocket.Chat Server and see if the issue is resolved.

As written above, I am using Rocket.Chat 6.1.1. This is the most recent version available.

> If none of these steps resolve the issue, you may want to reach out to the Rocket.Chat community for additional support or file a bug report if it is a software issue.

I _already filed a bug report_, that's why we are here.


@saurabhmj11 I highly suspect that your reply was created by some LLM like ChatGPT. If that is actually the case, I kindly ask you to refrain from that in the future. I don't think that comments like above are particularly helpful. Instead they are consuming time from everyone involved and do not move the issue towards being fixed.

**Comment by hugocostadev** [comment]
That issue comes from fuselage repo, in the message-parser package

![image](https://user-images.githubusercontent.com/20212776/230911562-228defc3-a005-4fab-8627-ee0480598c5b.png)
